library/src/main/java/com/novoda/downloadmanager/MigrationFactory.java

```diff
@@ -11,7 +13,24 @@ private MigrationFactory() {
         // Uses static methods.
     }
 
-    public static Migrator createVersionOneToVersionTwoMigrator(Context context, File databasePath, Migrator.Callback callback) {
+    public static MigrationServiceBinder migrationServiceBinder(Context context, final Migrator.Callback migrationCallback) {
+        final Handler mainThreadHandler = new Handler(Looper.getMainLooper());
+        Migrator.Callback mainThreadReportingMigrationCallback = new Migrator.Callback() {
```

Comment:
I figured this was the best place to hide the fact that the `Service` uses a background thread and we need to report back to the `main thread`.